xbitinfo/xbitinfo.py

```diff
@@ -376,7 +376,43 @@ def load_bitinformation(label):
         raise FileNotFoundError(f"No bitinformation could be found at {label+'.json'}")
 
 
-def get_keepbits(info_per_bit, inflevel=0.99):
+def get_trueKeepbits(info_per_bit, bitdim, threshold, tolerance, bit_vars):
+    cdf = _cdf_from_info_per_bit(info_per_bit, bitdim)
+    dimensions = info_per_bit.dims
+    dim_size = dimensions["dim"]
+    for var_name in bit_vars:
+        for i in range(dim_size):
+            infoArray = info_per_bit[var_name].isel(dim=i)
+            # total sum of information along a dimension
+            infSum = sum(infoArray).item()
+
+            # sum of first nine bits
+            first_Ninebits_sum = sum(infoArray[:9]).item()
+
+            cdf_array = cdf[var_name].isel(dim=i)
+            gradient_array = np.diff(cdf_array.values)
+
+            for i in range(9, len(gradient_array) - 1):
+                first_Ninebits_sum = first_Ninebits_sum + infoArray[i].item()
+                if (
+                    gradient_array[i]
+                ) < tolerance and first_Ninebits_sum >= threshold * infSum:
```

Comment:
`first_Ninebits_sum >= threshold * infSum`  is fixed for each iteration within this loop and will not change. You can therefor move this part of the condition outside of the `for` loop and skip this loop entirely if `first_Ninebits_sum >= threshold * infSum` is not met.

Reply:
If I am understanding your comment correctly ,I think you missed line 396 where I am updating the first_Ninebits_sum on every iteration. So it will keep changing on every iteration.

Reply:
True, you tricked me with the naming of `first_Ninebits_sum`, which sounds like it shouldn't change ;) maybe you can rename the variable or introduce an additional one.

Reply:
Yes I thought of changing it earlier too. Will do it now